File: js/0462-minimum-moves-to-equal-array-elements-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minMoves2 = function (nums) {
  nums.sort((a, b) => a - b);
  var mid = Math.floor(nums.length / 2);
  return nums
    .slice(0, mid)
    .reduce(
      (acc, curr, i) =>
        acc + (nums[mid] - curr) + (nums[nums.length - i - 1] - nums[mid]),
      0
    );
};

module.exports = { minMoves2 };

File: go/solutions/0462-minimum-moves-to-equal-array-elements-ii.go
package solutions

import "sort"

func MinMoves2(nums []int) int {
	sort.Ints(nums)
	n := len(nums)
	ans, mid := 0, nums[n/2]
	for i := 0; i < n/2; i++ {
		ans += mid - nums[i]
		ans += nums[n-i-1] - mid
	}
	return ans
}
